src/parsers/manifestjson.js

```diff
@@ -295,7 +295,7 @@ export default class ManifestJSONParser extends JSONParser {
     ) {
       // Choose a different message for permissions unsupported with the
       // add-on manifest_version.
-      if (PERMS_DATAPATH_REGEX.test(error.dataPath)) {
+      if (COMPLEX_ARRAYS_DATAPATH_REGEX.test(error.dataPath)) {
         baseObject = messages.manifestPermissionUnsupported(error.data, error);
```

Comment:
> @rpl are you OK with the change from "Unknown permission" to "Invalid permission"? (I don't think it is a problem personally)

@willdurand @diox I don't really mind changing "Unknown permission" into "Invalid permission", but I'd prefer if we could define a new `ORIGINS_DATAPATH_REGEX` (or `INSTALLORIGIN_DATAPATH_REGEX`) instead of repurposing the original `PERMS_DATAPATH_REGEX`.

As an example by reusing that pref here we are potentially going to be using `messages.manifestPermissionUnsupported` for something that isn't a permission anymore, even if it does not happen in this patch yet because at the moment we enter this branch only for the errors that also involve manifest_version based jsonschema checks (but it may become the case if in the future we may introduce some manifest_version-based changes for the install_origins field schema).

Reply:
@rpl would the following work for you ? :
```diff
diff --git a/src/const.js b/src/const.js
index 7602c4d1..42ff23ae 100644
--- a/src/const.js
+++ b/src/const.js
@@ -97,8 +97,9 @@ export const MAX_FILE_SIZE_TO_PARSE_MB = 4;
 export const HIDDEN_FILE_REGEX = /^__MACOSX\//;
 export const FLAGGED_FILE_REGEX = /thumbs\.db$|\.DS_Store$|\.orig$|\.old$|~$/i;
 export const ALREADY_SIGNED_REGEX = /^META-INF\/manifest\.mf/;
-export const COMPLEX_ARRAYS_DATAPATH_REGEX =
-  /^\/(permissions|optional_permissions|host_permissions|install_origins)\/([\d+])/;
+export const PERMS_DATAPATH_REGEX =
+  /^\/(permissions|optional_permissions|host_permissions)\/([\d+])/;
+export const INSTALLORIGIN_DATAPATH_REGEX = /^\/(install_origins)\/([\d+])/;
 
 export const RESERVED_FILENAMES = ['mozilla-recommendation.json'];
 
diff --git a/src/messages/manifestjson.js b/src/messages/manifestjson.js
index 48e1aeb7..8d89f744 100644
--- a/src/messages/manifestjson.js
+++ b/src/messages/manifestjson.js
@@ -1,7 +1,7 @@
 import { oneLine } from 'common-tags';
 
 import { i18n, errorParamsToUnsupportedVersionRange } from 'utils';
-import { MANIFEST_JSON, COMPLEX_ARRAYS_DATAPATH_REGEX } from 'const';
+import { MANIFEST_JSON, PERMS_DATAPATH_REGEX } from 'const';
 
 export const MANIFEST_FIELD_REQUIRED = {
   code: 'MANIFEST_FIELD_REQUIRED',
@@ -49,7 +49,7 @@ export function manifestPermissionUnsupported(permissionName, error) {
   const message = i18n.sprintf(messageTmpl, {
     permissionName,
     versionRange,
-    fieldName: error.dataPath.match(COMPLEX_ARRAYS_DATAPATH_REGEX)[1],
+    fieldName: error.dataPath.match(PERMS_DATAPATH_REGEX)[1],
   });
 
   return {
diff --git a/src/parsers/manifestjson.js b/src/parsers/manifestjson.js
index 260bb68c..371d0f7b 100644
--- a/src/parsers/manifestjson.js
+++ b/src/parsers/manifestjson.js
@@ -25,10 +25,11 @@ import {
   LOCALES_DIRECTORY,
   MESSAGES_JSON,
   FILE_EXTENSIONS_TO_MIME,
+  INSTALLORIGIN_DATAPATH_REGEX,
   STATIC_THEME_IMAGE_MIMES,
   RESTRICTED_HOMEPAGE_URLS,
   RESTRICTED_PERMISSIONS,
-  COMPLEX_ARRAYS_DATAPATH_REGEX,
+  PERMS_DATAPATH_REGEX,
 } from 'const';
 import log from 'logger';
 import * as messages from 'messages';
@@ -295,7 +296,7 @@ export default class ManifestJSONParser extends JSONParser {
     ) {
       // Choose a different message for permissions unsupported with the
       // add-on manifest_version.
-      if (COMPLEX_ARRAYS_DATAPATH_REGEX.test(error.dataPath)) {
+      if (PERMS_DATAPATH_REGEX.test(error.dataPath)) {
         baseObject = messages.manifestPermissionUnsupported(error.data, error);
       } else {
         baseObject = messages.manifestFieldUnsupported(error.dataPath, error);
@@ -333,7 +334,11 @@ export default class ManifestJSONParser extends JSONParser {
     // Arrays can be extremely verbose, this tries to make them a little
     // more sane. Using a regex because there will likely be more as we
     // expand the schema.
-    const match = error.dataPath.match(COMPLEX_ARRAYS_DATAPATH_REGEX);
+    // Note that this works because the 2 regexps use similar patterns. We'll
+    // want to adjust this if they start to differ.
+    const match =
+      error.dataPath.match(PERMS_DATAPATH_REGEX) ||
+      error.dataPath.match(INSTALLORIGIN_DATAPATH_REGEX);
 
     if (
       match &&
```

Reply:
@diox yep, that would be ok from my perspective too :+1:

Reply:
> +export const INSTALLORIGIN_DATAPATH_REGEX = /^\/(install_origins)\/([\d+])/;

I suppose we could add 2 more chars in the variable name to match the regexp: `INSTALLORIGIN_DATAPATH_REGEX` -> `INSTALL_ORIGINS_DATAPATH_REGEX